code/modules/bitrunning/objects/quantum_server.dm

```diff
@@ -0,0 +1,739 @@
+		if(accumulated_cost >= random_value)
+			domain_randomized = TRUE
+			return available["id"]
+
+/// Returns boolean if the server is ready to be used
+/obj/machinery/quantum_server/proc/get_is_ready()
+	return !loading && !cooling_off
+
+/// Gets all mobs originally generated by the loaded domain and returns a list that are capable of being antagged
+/obj/machinery/quantum_server/proc/get_valid_domain_targets()
+	// A: No one is playing
+	// B: The domain is not loaded
+	// C: The domain is shutting down
+	// D: There are no mobs
+	if(!length(occupant_mind_refs) || isnull(generated_domain) || !get_is_ready() || !length(mutation_candidates))
+		return list()
+
+	for(var/mob/living/creature as anything in mutation_candidates)
+		if(QDELETED(creature) || creature.mind)
+			mutation_candidates.Remove(creature)
+
+	return mutation_candidates
+
+/// Grades the player's run based on several factors
+/obj/machinery/quantum_server/proc/grade_completion(difficulty, threats, points, randomized, completion_time)
+	var/score = threats * 5
+	score += points
+	score += randomized ? 1 : 0
+
+	if(completion_time <= 2 MINUTES)
+		score += difficulty * 4
+	else if(completion_time <= 5 MINUTES)
+		score += difficulty * 3
+	else if (completion_time <= 10 MINUTES)
+		score += difficulty * 2
+	else
+		score += 1
+
+	switch(score)
+		if(1 to 4)
+			return "D"
+		if(5 to 7)
+			return "C"
+		if(8 to 10)
+			return "B"
+		if(11 to 13)
+			return "A"
+		else
+			return "S"
+
+/// Initializes a new domain if the given key is valid and the user has enough points
+/obj/machinery/quantum_server/proc/initialize_domain(map_key)
+	var/datum/lazy_template/virtual_domain/to_load
+
+	for(var/datum/lazy_template/virtual_domain/available as anything in subtypesof(/datum/lazy_template/virtual_domain))
+		if(map_key != initial(available.key) || points < initial(available.cost))
+			continue
+		to_load = available
+		break
+
+	if(isnull(to_load))
+		return FALSE
+
+	generated_domain = new to_load()
+	generated_domain.lazy_load()
+
+	return TRUE
+
+/// Loads in necessary map items, sets mutation targets, etc
+/obj/machinery/quantum_server/proc/initialize_map_items()
+	for(var/thing in generated_domain.created_atoms_list)
+		if(isliving(thing)) // so we can mutate them
+			var/mob/living/creature = thing
+			if(creature.can_be_cybercop)
+				mutation_candidates += creature
+
+			continue
+
+		if(istype(thing, /obj/effect/mob_spawn/ghost_role)) // so we get threat alerts
+			RegisterSignal(thing, COMSIG_GHOSTROLE_SPAWNED, PROC_REF(on_threat_created))
+			continue
+
+		if(istype(thing, /obj/effect/mob_spawn/corpse)) // corpses are valid targets too
+			var/obj/effect/mob_spawn/corpse/spawner = thing
+			var/mob/living/creature = spawner.mob_ref?.resolve()
+
+			if(!creature?.can_be_cybercop)
+				continue
+
+			mutation_candidates += creature
+			qdel(spawner)
+
+	var/turf/goal_turfs = list()
+	for(var/obj/thing in generated_safehouse.created_atoms)
```

Comment:
```suggestion
	for(var/obj/thing as anything in generated_safehouse.created_atoms)
```
If all created atoms are objects, then you can put `as anything` here to skip the typecheck. Small micro-opt but resolve this change if it can contain non-objects too.

Reply:
it's all atoms